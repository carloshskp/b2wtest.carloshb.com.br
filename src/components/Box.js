import React  from 'react';
import PropTypes from 'prop-types';

import Label from './Label';

const Box = ({ planet, population, climate, terrain, featured }) => <section className="box">
    <header>
        <h1>{ planet }</h1>
    </header>
    <div className="content">
        <Label title='POPULATION'>{ population }</Label>
        <Label title='CLIMATE'>{ climate }</Label>
        <Label title='TERRAIN'>{ terrain }</Label>
    </div>
    <footer>
        Featured in { featured } films.
    </footer>
</section>;

Box.propTypes = {
    planet: PropTypes.string.isRequired,
    population: PropTypes.string.isRequired,
    climate: PropTypes.string.isRequired,
    terrain: PropTypes.string.isRequired,
    featured: PropTypes.number.isRequired,
};

export default Box;