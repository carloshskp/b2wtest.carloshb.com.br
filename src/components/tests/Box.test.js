import React from 'react';
import renderer from 'react-test-renderer';

import Box from './Box';

it('renders Box component', () => {
    const box = renderer.create(
        <Box
            planet='PLANET NAME'
            population='200000'
            climate='ARID'
            terrain='DESERT'
            featured={2}
        />
    );

    let tree = box.toJSON();

    expect(tree).toMatchSnapshot();
});
