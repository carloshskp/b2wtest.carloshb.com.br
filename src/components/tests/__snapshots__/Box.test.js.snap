// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Box component 1`] = `
<section
  className="box undefined"
>
  <header>
    <h1>
      PLANET NAME
    </h1>
  </header>
  <div
    className="content"
  >
    <span
      className="label"
    >
      <strong>
        POPULATION
      </strong>
      : 
      200000
    </span>
    <span
      className="label"
    >
      <strong>
        CLIMATE
      </strong>
      : 
      ARID
    </span>
    <span
      className="label"
    >
      <strong>
        TERRAIN
      </strong>
      : 
      DESERT
    </span>
  </div>
  <footer>
    Featured in 
    2
     films.
  </footer>
</section>
`;
